uaclient/tests/test_log.py

```diff
@@ -158,3 +159,42 @@ def test_valid_json_output(
             assert val[6].get("key") == extra.get("key")
         else:
             assert 7 == len(val)
+
+
+class TestLogHelpers:
+    def test_get_user_or_root_log_file_path(self):
```

Comment:
This looks like a good set of tests, but let's use parametrization and mock decorators to follow the patterns in the rest of the codebase. Something like this:
```suggestion
    @pytest.mark.parametrize(
        [
            "we_are_currently_root",
            "expected",
        ],
        [
            (True, "cfg_log_file"),
            (False, "user_log_file"),
        ]
    )
    @mock.patch("uaclient.util.we_are_currently_root")
    @mock.patch("uaclient.config.UAConfig.log_file", new_callable=mock.PropertyMock, return_value="cfg_log_file")
    @mock.patch("uaclient.log.get_user_log_file", return_value="user_log_file")
    def test_get_user_or_root_log_file_path(
        self,
        m_we_are_currently_root,
        m_cfg_log_file,
        m_get_user_log_file,
        we_are_currently_root,
        expected
    ):
```

Reply:
@a-dubs sorry, I just noticed a confusing error in my suggestion above - the mock decorators were in the wrong order. It should be fixed now

Reply:
thank you! will implement these changes now. Will break the last portion of the test out into a different function because it does not make sense with the parameterization.

Reply:
nevermind I'm just going to get rid of that test case because it is redundant.